kolibri/content/models.py

```diff
@@ -212,80 +124,73 @@ class Admin:
     def __str__(self):
         return self.license_name
 
-class ContentRelationship(AbstractContent):
+class PrerequisiteContentRelationship(AbstractContent):
     """
-    Provide a abstract model for defining any relationships between two ContentMetadata objects.
+    Predefine the prerequisite relationship between two ContentNode objects.
     """
-    contentmetadata_1 = models.ForeignKey(ContentMetadata, related_name='%(app_label)s_%(class)s_1')
-    contentmetadata_2 = models.ForeignKey(ContentMetadata, related_name='%(app_label)s_%(class)s_2')
-
-    class Meta:
-        abstract = True
+    target_node = models.ForeignKey(ContentNode, related_name='%(app_label)s_%(class)s_target_node')
+    prerequisite = models.ForeignKey(ContentNode, related_name='%(app_label)s_%(class)s_prerequisite')
 
-    class Admin:
-        pass
-
-class PrerequisiteContentRelationship(ContentRelationship):
-    """
-    Predefine the prerequisite relationship between two ContentMetadata objects.
-    """
     class Meta:
-        unique_together = ['contentmetadata_1', 'contentmetadata_2']
+        unique_together = ['target_node', 'prerequisite']
 
     class Admin:
         pass
 
     def clean(self, *args, **kwargs):
         # self reference exception
-        if self.contentmetadata_1 == self.contentmetadata_2:
+        if self.target_node == self.prerequisite:
             raise IntegrityError('Cannot self reference as prerequisite.')
         # immediate cyclic exception
         elif PrerequisiteContentRelationship.objects.using(self._state.db)\
-                .filter(contentmetadata_1=self.contentmetadata_2, contentmetadata_2=self.contentmetadata_1):
+                .filter(target_node=self.prerequisite, prerequisite=self.target_node):
```

Comment:
Just a note: doing DB queries whenever we save (as `save` calls `full_clean`) will be slow. How often do we make changes to a `PrerequisiteContentRelationship`?